### PR TITLE
Spawning multiple goroutine for Notification and Sharing http client

### PR DESCRIPTION
**Pull request by saurabh2253**
This fixes the case when there are all integrations configured and few were not getting processed.

Fixes #
https://github.com/deepfence/enterprise-roadmap/issues/1943

Changes proposed in this pull request:
- Add timeout to http requests
- Spawn a goroutine for each integration
- Share http client across all goroutines.

### CODE REVIEW

File: deepfence_worker/cronjobs/notification.go [hunk 1]
```diff
@@ -30,24 +30,34 @@ func SendNotifications(msg *message.Message) error {
...
-			processIntegration[model.Secret](msg, integrationRow)
-		case utils.ScanTypeDetectedNode[utils.NEO4J_MALWARE_SCAN]:
-			processIntegration[model.Malware](msg, integrationRow)
-		case utils.ScanTypeDetectedNode[utils.NEO4J_COMPLIANCE_SCAN]:
-			processIntegration[model.Compliance](msg, integrationRow)
-			// cloud compliance scans
-			integrationRow.Resource = utils.ScanTypeDetectedNode[utils.NEO4J_CLOUD_COMPLIANCE_SCAN]
-			processIntegration[model.CloudCompliance](msg, integrationRow)
-		}
+		go processIntegrationRow(wg, integrationRow, msg)
```
Comment (noboruma) [inline deepfence_worker/cronjobs/notification.go:36]: I am not sure we need to do this at all.
What is this solving?

Comment (noboruma) [reply deepfence_worker/cronjobs/notification.go:36]: I would go one step further and limit to 1, removing `go` routine entirely ;)
Unless we have a good reason, here it feels we are making the code more complex for little benefit, we are actually stressing the DB access & memory usage more than necessary.
Instead we should optimize the sending of HTTP requests (batch + go routines)

Comment (gnmahanth) [reply deepfence_worker/cronjobs/notification.go:36]: ok that should minimize frequent neo4j access

**Comment by noboruma** [comment]
I have found a couple of issues (memory leak, broken batching) in the current code base.
Let's merge this in and revamp the code to make it more generic & fix those issues

**Pull request merged**